aruco_ros/CMakeLists.txt

```diff
@@ -23,98 +21,112 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
     sensor_msgs
     visualization_msgs
 )
+
 find_package(ament_cmake REQUIRED)
-foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
-  find_package(${Dependency} REQUIRED)
+foreach(DEP ${dependencies})
+  find_package(${DEP} REQUIRED)
 endforeach()
 
-#generate_dynamic_reconfigure_options(
-#  cfg/ArucoThreshold.cfg
-#)
-
-add_library(aruco_ros_utils SHARED src/aruco_ros_utils.cpp)
-target_include_directories(aruco_ros_utils
+# Build library.
+add_library(${PROJECT_NAME}_utils SHARED
+  src/aruco_ros_utils.cpp
+)
+target_include_directories(${PROJECT_NAME}_utils
   PUBLIC
-  include)
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include/${PROJECT_NAME}>)
```

Comment:
This should only be:
```cmake
$<INSTALL_INTERFACE:include>
```
As you are supposed to install the *content* of the `<package_name>/include` folder in the `include` destination.

This way the users will have to include: `#include "project_name/my_headers.hpp"` which is crystal clear.